Add Monit checks and alerts templates

morrismukiri opened the pull request:
Add template scripts to enable checking SMPP connection status and send alerts via Email and OpsGenie.

morrismukiri committed:
Add Monit checks and alerts templates

Add template scripts to enable checking SMPP connection status
and send alert via Email and OpsGenie.

Signed-off-by: Morris Mukiri <morrismukiri@gmail.com>

morrismukiri committed:
Add Tasks to copy Monit scripts

Signed-off-by: Morris Mukiri <morrismukiri@gmail.com>

jasonrogena commented on tasks/monitoring.yml:
```diff
@@ -0,0 +1,25 @@
+
+- name: Make sure monit exec-scripts directory exists
+  file:
+    path: "/etc/monit/exec-sripts/"
```
@morrismukiri there seems to be a typo in the path. Should probably be `/etc/monit/exec-scripts` right?

jasonrogena commented on templates/etc/monit/conf.d/smpp-host-connection.j2:
```diff
@@ -0,0 +1,8 @@
+        port {{ kannel_monit_check_host_port }}
+        type {{ kannel_monit_check_host_type }}
+    THEN
+      exec "/etc/monit/exec-scripts/ipsec-action failed"
```
Should this be `/etc/monit/exec-scripts/smpp-connection-alerts.sh failed` instead?

jasonrogena commented on templates/etc/monit/conf.d/smpp-host-connection.j2:
```diff
@@ -0,0 +1,8 @@
+    THEN
+      exec "/etc/monit/exec-scripts/ipsec-action failed"
+    ELSE IF SUCCEEDED for {{ kannel_monit_check_times }} times in {{ kannel_monit_check_cycles }} cycles THEN
+      exec "/etc/monit/exec-scripts/ipsec-action success"
```
Should this be `/etc/monit/exec-scripts/smpp-connection-alerts.sh success` instead?

jasonrogena commented on templates/etc/monit/exec-scripts/smpp-connection-alerts.sh.j2:
```diff
@@ -0,0 +1,51 @@
+    echo "$MONIT_DATE" > "$ipsec_failure_file"
+    # Send first incident occurrence alert
+    # Create Opsgenie Incident and save request ID to file
+    source /etc/monit/exec-scripts/opsgenie-alert
```
Shoud this be `source /etc/monit/exec-scripts/smpp-connection-opsgenie-alerts.sh` instead?

jasonrogena commented on templates/etc/monit/exec-scripts/smpp-connection-alerts.sh.j2:
```diff
@@ -0,0 +1,51 @@
+    time_difference=$(expr ${current_time_epoch} - ${first_incident_time_epoch})
+    incident_duration=$(date '+%H hours, %M minutes and %S seconds' -ud @${time_difference})
+    # Close Opsgenie incident
+    source /etc/monit/exec-scripts/opsgenie-alert
```
Shoud this be `source /etc/monit/exec-scripts/smpp-connection-opsgenie-alerts.sh` instead?

jasonrogena commented on templates/etc/monit/exec-scripts/smpp-connection-alerts.sh.j2:
```diff
@@ -0,0 +1,51 @@
+        -F subject="$(hostname): {{ kannel_monit_check_host_name }} SMPP Connection" \
+        -F text="The {{ kannel_monit_check_host_name }} SMPP connection in the host $(hostname) has been lost from around '$MONIT_DATE UTC'!"
+
+    /etc/monit/exec-scripts/slack-notification
```
```suggestion
```

jasonrogena commented on templates/etc/monit/exec-scripts/smpp-connection-alerts.sh.j2:
```diff
@@ -0,0 +1,51 @@
+    # Clear failure incident file content
+    :> "$ipsec_failure_file"
+
+    /etc/monit/exec-scripts/slack-notification
```
```suggestion
```

morrismukiri committed:
Move file paths to variable file

Signed-off-by: Morris Mukiri <morrismukiri@gmail.com>

morrismukiri committed:
update Molecule tests

Update molecule syntax
fix linting and tests
add monit as a dependency

Signed-off-by: Morris Mukiri <morrismukiri@gmail.com>

jasonrogena commented on molecule/default/molecule.yml:
```diff
@@ -3,20 +3,31 @@ dependency:
-lint:
-  name: yamllint
+lint: |
+  yamllint .
```
Wonder if the lint step will fail without us specifying `set -e` as the first line.

jasonrogena commented on meta/main.yml:
```diff
@@ -17,4 +18,11 @@ galaxy_info:
+      monit_setup_mode: "{{ kannel_monit_setup_mode }}"
+      monit_scripts: "{{ kannel_monit_scripts }}"
+      monit_opsgenie_api_key: "{{ kannel_monit_opsgenie_api_key }}"
+      slack_monit_endpoint: "{{ kannel_slack_monit_endpoint }}"
```
@morrismukiri since the usefulness of the monitoring scripts is pegged on how often Monit should be running them (currently 20 seconds), I think we should also pass `monit_check_interval` with a default value of `20`.

morrismukiri committed:
Add Monit check interval

Modify the interval at which mnit checks the SMPP connection
and set default to 20 seconds.

Signed-off-by: Morris Mukiri <morrismukiri@gmail.com>
